File: ui/react/src/action/RenameAsset.ts
import SheetAction from "./SheetAction";
import Action from "./Action";

class RenameAsset extends SheetAction {
    public oldAssetName: string;
    public newAssetName: string;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const oldAssetName: string = input.oldAssetName;
        const newAssetName: string = input.newAssetName;
        return new RenameAsset(sheetName, oldAssetName, newAssetName);
    }

    constructor(
        sheetName: string,
        oldAssetName: string,
        newAssetName: string) {

        super(sheetName);
        this.oldAssetName = oldAssetName;
        this.newAssetName = newAssetName;
    }

    public isAssetAction() {
        return true;
    }

    public targetAsset() {
        return this.oldAssetName;
    }

    public wrapper(): Action {
        const action = new Action();
        action.renameAsset = this;
        return action;
    }

}

export default RenameAsset;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/RenameAsset.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.api.action.Action;

public class RenameAsset extends SheetAction implements Action {
    private String oldAssetName;
    private String newAssetName;

    public RenameAsset() {
    }

    public RenameAsset(String sheetName, String oldAssetName, String newAssetName) {
        super(sheetName);
        this.oldAssetName = oldAssetName;
        this.newAssetName = newAssetName;
    }

    public String getOldAssetName() {
        return oldAssetName;
    }

    public void setOldAssetName(String oldAssetName) {
        this.oldAssetName = oldAssetName;
    }

    public String getNewAssetName() {
        return newAssetName;
    }

    public void setNewAssetName(String newAssetName) {
        this.newAssetName = newAssetName;
    }
}
